modules/lib/src/main/resources/assets/js/app/browse/action/MarkAsReadyContentAction.ts

```diff
@@ -37,8 +36,49 @@ export class MarkAsReadyContentAction
         if (isSingleItem) {
             this.markAsReadyAndPublish(contentToMarkAsReady, content);
         } else {
-            this.confirmDialog.setYesCallback(() => void this.markAsReadyAndPublish(contentToMarkAsReady, content)).open();
+            // Mount a UI2 ConfirmationDialog on demand
+            const mount = document.createElement('div');
+            document.body.appendChild(mount);
+
+            const cleanup = (): void => {
+                render(null, mount);
+                mount.remove();
+            };
```

Comment:
I fully understand why you did this. I just think we can do better, even for a temporary solution. What if you export `class ConfirmationDialogElement extends LegacyElement` and use that element here? That dialog will already present the root to render dialog. You can implement `open` and `close` functions, like in old modals, and just set the `open` state instead.